Provider/Pool.php

```diff
@@ -45,8 +45,14 @@ public function __construct($context)
      */
     public function getProvider($name)
     {
+        if (!$name) {
+            throw new \InvalidArgumentException('Provider name cannot be empty, did you forget to call setProviderName() in your Media object?');
+        }
+        if (empty($this->providers)) {
+            throw new \RuntimeException('There are no providers configured yet');
+        }
         if (!isset($this->providers[$name])) {
-            throw new \RuntimeException(sprintf('unable to retrieve the provider named : `%s`', $name));
+            throw new \InvalidArgumentException(sprintf('Unable to retrieve the provider named "%s". Available providers are %s.', $name, '"'.implode('", "', $this->getProviderList()).'"'));
```

Comment:
Yes. This is exactly why I said to put `"` on sprintf. Something like this:

```php
throw new \InvalidArgumentException(sprintf('Unable to retrieve the provider named "%s". Available providers are "%s".', $name, implode('", "', $this->getProviderList())));
```

Much proper IMO.